### PR TITLE
DAOS-13895 cq: Fix complex log/error lines that do not contain newline

### PR DESCRIPTION
**Pull request by ashleypittman**
Fix log and error messages so that they contain at least one newline.
Remove some logging that's not needed.

Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]

**Review by knard-intel — COMMENTED** [review]
Mostly OK for me.
Do not understand why some error messages have been removed.
However, I am probably missing something as I am not used to this part of the code.

File: src/engine/drpc_listener.c [hunk 1]
```diff
@@ -143,12 +143,9 @@ drpc_listener_start_ult(ABT_thread *thread)
 static int
 generate_socket_path(void)
 {
-	D_ASPRINTF(drpc_listener_socket_path, "%s/daos_engine_%d.sock",
-		   dss_socket_dir, getpid());
-	if (drpc_listener_socket_path == NULL) {
-		D_ERROR("Failed to allocate socket path\n");
+	D_ASPRINTF(drpc_listener_socket_path, "%s/daos_engine_%d.sock", dss_socket_dir, getpid());
+	if (drpc_listener_socket_path == NULL)
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/engine/drpc_listener.c:147]: Our allocation macros will all log if there's a failure anyway, this error would be a duplicate although with slightly different text.

[the remark this answers is not in the document]
Comment (knard-intel) [reply src/engine/drpc_listener.c:147]: OK, thanks for the explanation.

File: src/rdb/rdb.c [hunk 2]
```diff
@@ -245,8 +245,6 @@ rdb_open_internal(daos_handle_t pool, daos_handle_t mc, const uuid_t uuid, uint6
 
 	D_ALLOC_PTR(db);
 	if (db == NULL) {
-		D_ERROR(DF_UUID": failed to allocate db object\n",
```
[the remark this answers is not in the document]
Comment (knard-intel) [reply src/rdb/rdb.c:248]: OK

**Review by knard-intel — DISMISSED** [review]

File: src/rdb/rdb.c [hunk 3]
```diff
@@ -245,6 +245,8 @@ rdb_open_internal(daos_handle_t pool, daos_handle_t mc, const uuid_t uuid, uint6
 
 	D_ALLOC_PTR(db);
 	if (db == NULL) {
+		D_ERROR(DF_UUID": failed to allocate db object\n",
+			DP_UUID(uuid));
```
Comment (daltonbohning) [inline src/rdb/rdb.c]: Does the error not need to print DER_NOMEM?

**Review by daltonbohning — APPROVED** [review]

File: src/rdb/rdb.c (hunk 3 above)
Comment (ashleypittman) [reply src/rdb/rdb.c]: I've reverted this change but yes it should.  Out of scope though.

**Pull request merged**